x-pack/test/security_solution_cypress/cypress/e2e/detection_response/rule_management/coverage_overview/coverage_overview.cy.ts

```diff
@@ -0,0 +1,429 @@
+        cy.get(COVERAGE_OVERVIEW_POPOVER_DISABLED_RULES)
+          .contains('Disabled prebuilt rule')
+          .should('not.exist');
+
+        openTechniquePanelByName(EnabledCustomRuleMitreData.technique.name);
+        cy.get(COVERAGE_OVERVIEW_POPOVER_ENABLED_RULES).contains('Enabled custom rule');
+
+        openTechniquePanelByName(DisabledCustomRuleMitreData.technique.name);
+        cy.get(COVERAGE_OVERVIEW_POPOVER_DISABLED_RULES)
+          .contains('Disabled custom rule')
+          .should('not.exist');
+      });
+    });
+
+    it('enables all disabled rules', () => {
+      selectCoverageOverviewActivityFilterOption('Disabled rules');
+      openTechniquePanelByName(DisabledPrebuiltRuleMitreData.technique.name);
+      enableAllDisabledRules();
+
+      // Should now render all rules in "enabled" section
+      openTechniquePanelByName(DisabledPrebuiltRuleMitreData.technique.name);
+      cy.get(COVERAGE_OVERVIEW_POPOVER_ENABLED_RULES).contains('Disabled prebuilt rule');
+
+      // Shouldn't render the rules in "disabled" section
+      cy.get(COVERAGE_OVERVIEW_POPOVER_DISABLED_RULES)
+        .contains('Disabled prebuilt rule')
+        .should('not.exist');
+      cy.get(COVERAGE_OVERVIEW_ENABLE_ALL_DISABLED_BUTTON).should('be.disabled');
+    });
+  });
+
+  describe('with rules that have identical mitre techniques that belong to multiple tactics', () => {
+    beforeEach(() => {
+      login();
+      deleteAlertsAndRules();
+      deletePrebuiltRulesAssets();
+      createRule(
+        getNewRule({
+          rule_id: 'duplicate_technique_rule_1',
+          enabled: true,
+          name: 'Rule with tactic 1',
+          threat: [MockCustomRuleDuplicateTechniqueThreat1],
+        })
+      );
+      createRule(
+        getNewRule({
+          rule_id: 'duplicate_technique_rule_2',
+          name: 'Rule with tactic 2',
+          enabled: false,
+          threat: [MockCustomRuleDuplicateTechniqueThreat2],
+        })
+      );
+      visit(RULES_COVERAGE_OVERVIEW_URL);
+    });
+
+    it('technique panels render unique rule data', () => {
+      selectCoverageOverviewActivityFilterOption('Disabled rules');
+
+      // Open duplicated technique panel under first tactic
+      cy.get(COVERAGE_OVERVIEW_TECHNIQUE_TITLE(DuplicateTechniqueMitreData1.technique.id))
+        .first()
+        .click();
+      cy.get(COVERAGE_OVERVIEW_POPOVER_ENABLED_RULES)
+        .contains('Rule with tactic 1')
+        .should('not.exist');
+      cy.get(COVERAGE_OVERVIEW_POPOVER_DISABLED_RULES).contains('Rule with tactic 2');
+
+      // Open duplicated technique panel under second tactic
+      cy.get(COVERAGE_OVERVIEW_TECHNIQUE_TITLE(DuplicateTechniqueMitreData2.technique.id))
+        .last()
+        .click();
+      cy.get(COVERAGE_OVERVIEW_POPOVER_ENABLED_RULES).contains('Rule with tactic 1');
+      cy.get(COVERAGE_OVERVIEW_POPOVER_DISABLED_RULES)
+        .contains('Rule with tactic 2')
+        .should('not.exist');
+    });
```

Comment:
Somehow I couldn't parse this even after running this test locally and checking what's going on in the app. Could you please add comments to every instruction explaining what do we check here?

Reply:
Just a few examples what is totally unclear w/o debugging it under a 🔍 

- `cy.get(COVERAGE_OVERVIEW_TECHNIQUE_TITLE(DuplicateTechniqueMitreData1.technique.id)).first()` - this should correspond to the shared technique, but to which one of the two? If the test could specify the tactic first, and then try to find the technique under this tactic, it would be much clearer.
- `DuplicateTechniqueMitreData1.technique.id`, `DuplicateTechniqueMitreData2.technique.id` - numbers don't help understand the code, it would be easier to remember that "rule one is mapped to Event Triggered Execution via Privilege Escalation, and rule 2 is mapped to it via Persistence"
- Maybe names like `sharedTechniqueId`, `tacticIdOfRule1`, `tacticIdOfRule2` could help

Reply:
I can be a bit more descriptive with the names and maybe put the repeated actions in a labeled task along with comments. Basically, we're unable to retrieve the technique panels the same way we do the rest of the times in this test because they have the same name and cypress only returns the first option when using the `.contains` method so it looks a bit funky compared to the other tests.